plugins/ltac/tacarg.ml

```diff
@@ -19,13 +19,19 @@ let make0 ?dyn name =
   let () = Geninterp.register_val0 wit dyn in
   wit
 
-let wit_intro_pattern = make0 "intropattern"
+let wit_intropattern = make0 "intropattern" (* To keep after deprecation phase but it will get a different parsing semantics (Tactic Notation and TACTIC EXTEND) in pltac.ml *)
+let wit_simple_intropattern = make0 "simple_intropattern"
 let wit_quant_hyp = make0 "quant_hyp"
 let wit_constr_with_bindings = make0 "constr_with_bindings"
 let wit_open_constr_with_bindings = make0 "open_constr_with_bindings"
 let wit_bindings = make0 "bindings"
 let wit_quantified_hypothesis = wit_quant_hyp
-let wit_intropattern = wit_intro_pattern
+
+(* A convenient common part to simple_intropattern and intropattern
+   usable when no parsing rule is concerned: indeed
+   simple_intropattern and intropattern are in the same type and have
+   the same interp/intern/subst methods *)
+let wit_intro_pattern = wit_intropattern
```

Comment:
Forgot to mention: it's probably better to share the same runtime representation by setting the `dyn` flag. Not critical though.

Reply:
Ok, done.